datalad/tests/test_dataset.py

```diff
@@ -83,3 +95,84 @@ def test_Dataset_get(src, dst):
     assert_equal(f.readlines(), ['123\n'], "test-annex.dat's content doesn't match.")
 
     os.chdir(cwd)
+
+
+@assert_cwd_unchanged
+@with_testrepos(flavors=local_flavors)
+@with_tempfile
+def test_Dataset_add_to_annex(src, dst):
+
+    ds = Dataset(dst, src)
+    filename = 'file_to_annex.dat'
+    filename_abs = os.path.join(dst, filename)
+    f = open(filename_abs, 'w')
+    f.write("What to write?")
+    f.close()
+    cwd = os.getcwd()
+    os.chdir(dst)
+    ds.add_to_annex([filename])
+    os.chdir(cwd)
+
+    if not ds.is_direct_mode():
+        assert_true(os.path.islink(filename_abs), "Annexed file is not a link.")
+        ok_clean_git(dst, annex=True)
+    else:
+        assert_false(os.path.islink(filename_abs), "Annexed file is link in direct mode.")
+        # TODO: How to test the file was added in direct mode?
+        # May be this will need 'git annex find' or sth. to be implemented.
+        ok_clean_git_annex_proxy(dst)
+
+
+@assert_cwd_unchanged
+@with_testrepos(flavors=local_flavors)
+@with_tempfile
+def test_Dataset__add_to_git(src, dst):
+
+    ds = Dataset(dst, src)
+    # For now adding to git in direct doesn't seem to work
+    if ds.is_direct_mode():
+        raise SkipTest
+
+    filename = 'file_to_git.dat'
+    filename_abs = os.path.join(dst, filename)
+    f = open(filename_abs, 'w')
+    f.write("What to write?")
+    f.close()
+    cwd = os.getcwd()
+    os.chdir(dst)
+
+    ds.add_to_git([filename_abs])
+
+    if ds.is_direct_mode():
+        ok_clean_git_annex_proxy(dst)
+    else:
+        ok_clean_git(dst, annex=False)
+    os.chdir(cwd)
+
+
+@assert_cwd_unchanged
+@with_testrepos(flavors=local_flavors)
+@with_tempfile
+def test_Dataset_commit(src, path):
+
+    ds = Dataset(path, src)
+    filename = os.path.join(path, "test_git_add.dat")
+    f = open(filename, 'w')
+    f.write("File to add to git")
+    f.close()
+    cwd = os.getcwd()
+    os.chdir(path)
```

Comment:
since it seems to become a common pattern, I feel we better tune @assert_cwd_unchanged with option (e.g. "ok_to_chdir=False") which if True would just change back without raising AssertionError (unless there was already one raised by the test)... will do that now